src/client.js

```diff
@@ -0,0 +1,93 @@
+const formatsByCoinType = require('@ensdomains/address-encoder').formatsByCoinType;
+const Command = require('commander').Command;
+const ethers =  require('ethers');
+const OffchainResolver_abi = require('@ensdomains/offchain-resolver-contracts/artifacts/contracts/OffchainResolver.sol/OffchainResolver.json').abi;
+const IResolverService_abi = require('@ensdomains/offchain-resolver-contracts/artifacts/contracts/OffchainResolver.sol/IResolverService.json').abi;
+const Resolver_abi = require('@ensdomains/ens-contracts/artifacts/contracts/resolvers/Resolver.sol/Resolver.json').abi;
+
+const IResolverService = new ethers.utils.Interface(IResolverService_abi);
+
+function getDnsName(name) {
+  const n = name.replace(/^\.|\.$/gm, '');
+
+  var bufLen = n === '' ? 1 : n.length + 2;
+  var buf = Buffer.allocUnsafe(bufLen);
+
+  let offset = 0;
+  if (n.length) {
+    const list = n.split('.');
+    for (let i = 0; i < list.length; i++) {
+      const len = buf.write(list[i], offset + 1);
+      buf[offset] = len;
+      offset += len + 1;
+    }
+  }
+  buf[offset++] = 0;
+  return (
+    '0x' +
+    buf.reduce(
+      (output, elem) => output + ('0' + elem.toString(16)).slice(-2),
+      ''
+    )
+  );
+}
+
+const program = new Command();
+program
+  .requiredOption('-r --registry <address>', 'ENS registry address')
+  .option('-p --provider <url>', 'web3 provider URL', 'http://localhost:8545/')
+  .option('-i --chainId <chainId>', 'chainId', '1337')
+  .option('-n --chainName <name>', 'chainName', 'unknown')
+  .argument('<name>');
+
+program.parse(process.argv);
+
+const options = program.opts();
+const ensAddress = options.registry;
+const chainId = parseInt(options.chainId);
+const chainName = options.chainName;
+const provider = new ethers.providers.JsonRpcProvider(options.provider, {
+  chainId,
+  name: chainName,
+  ensAddress,
+});
+(async () => {
```

Comment:
This seems to replicate a lot of the functionality of the universal resolver. Why not just use that?